src/main.js

```diff
@@ -496,20 +507,34 @@ class MiniGraphCard extends LitElement {
 
   renderSvg() {
     const { height } = this.config;
+
+    const orderedKeys = [...this.fill.keys()];
+
+    // sort keys of elements by their graph order (ascending)
+    orderedKeys.sort((a, b) => this.Graph[b].order - this.Graph[a].order);
+
+    // map element keys to render function
+    const fillMap = orderedKeys.map((i, order) => this.renderSvgFill(this.fill[i], i, order));
+    const fillRectMap = orderedKeys.map(i => this.renderSvgFillRect(this.fill[i], i));
+    const lineMap = orderedKeys.map((i, order) => this.renderSvgLine(this.line[i], i, order));
+    const lineRectMap = orderedKeys.map(i => this.renderSvgLineRect(this.line[i], i));
+    const barMap = orderedKeys.map(i => this.renderSvgBars(this.bar[i], i));
+    const pointsMap = orderedKeys.map((i, order) => this.renderSvgPoints(this.points[i], i, order));
+
     return svg`
       <svg width='100%' height=${height !== 0 ? '100%' : 0} viewBox='0 0 500 ${height}'
         @click=${e => e.stopPropagation()}>
         <g>
           <defs>
             ${this.renderSvgGradient(this.gradient)}
           </defs>
-          ${this.fill.map((fill, i) => this.renderSvgFill(fill, i))}
-          ${this.fill.map((fill, i) => this.renderSvgFillRect(fill, i))}
-          ${this.line.map((line, i) => this.renderSvgLine(line, i))}
-          ${this.line.map((line, i) => this.renderSvgLineRect(line, i))}
-          ${this.bar.map((bars, i) => this.renderSvgBars(bars, i))}
+          ${fillMap}
+          ${fillRectMap}
+          ${lineMap}
+          ${lineRectMap}
+          ${barMap}
         </g>
-        ${this.points.map((points, i) => this.renderSvgPoints(points, i))}
+        ${pointsMap}
```

Comment:
While we might want to rework this, not ever map is based on `this.fill` (`show_fill` can be set to false); there's also no real benefit moving these to `const`'s as it's the only usage, and the function is returning svg elements

Reply:
Honestly, I made these changes in about two days and I can only remember pieces of the code. I'm not in the mindset for deep discussions about details right now. If you have a better solution I'm happy to adjust my code!

Reply:
This comment was not meant as a deep discussion; the proposed change would skip entities where `show_fill` is set to false. But let's park this until we have a decision regarding the `order` attribute in general (and again, thanks for your contribution regardless)